### DISCUSSION TITLE
Raycast Vehicle coordinate system confusion and wheel rotation issues

### DISCUSSION THREAD

**Discussion by aardvarkk** [bulletphysics/bullet3#3373] [General]
I have [filed a PR](https://github.com/bulletphysics/bullet3/pull/3282) to fix this issue that goes into greater detail about the setup, but I wanted to file an issue as well in case other people run into this and want a place to look.

I believe the issue I was having is the same as the one [documented here](https://www.gamedev.net/forums/topic/704633-bullet-raycast-vehicle-understanding-coordinate-system/).

**Comment by stephengold** [comment]
There's definitely confusion around vehicle/wheel coordinate systems. Some of it may stem from how easy it is to unintentionally configure a left-handed coordinate system using `btRaycastVehicle::setCoordinateSystem()`.

Instead of reversing signs yet again, I think a better way forward would be to define a (right,forward,up) basis for vehicle coordinates that's always right-handed. That would clarify correct usage of `btRaycastVehicle` while increasing its flexibility. I don't see any reason why a vehicle's forward direction *must* coincide with one of the local axes of its rigid body...
